`test_ttl_move/test.py::test_alter_with_merge_work` is flaky

https://s3.amazonaws.com/clickhouse-test-reports/56493/4ac3dcc417bb629c9731aa7447d030aabf56b5a3/integration_tests__tsan__[5_6].html

CI DB: [link](https://play.clickhouse.com/play?user=play#CldJVEgKICAgICd0ZXN0X2FsdGVyX3dpdGhfbWVyZ2Vfd29yaycgQVMgbmFtZV9zdWJzdHIsCiAgICA5MCBBUyBpbnRlcnZhbF9kYXlzLAogICAgKCdTdGF0ZWxlc3MgdGVzdHMgKGFzYW4pJywgJ1N0YXRlbGVzcyB0ZXN0cyAoYWRkcmVzcyknLCAnU3RhdGVsZXNzIHRlc3RzIChhZGRyZXNzLCBhY3Rpb25zKScpIEFTIGJhY2twb3J0X2FuZF9yZWxlYXNlX3NwZWNpZmljX2NoZWNrcwpTRUxFQ1QKICAgIHRvU3RhcnRPZkRheShjaGVja19zdGFydF90aW1lKSBBUyBkLAogICAgY291bnQoKSwKICAgIGdyb3VwVW5pcUFycmF5KHB1bGxfcmVxdWVzdF9udW1iZXIpIEFTIHBycywKICAgIGFueShyZXBvcnRfdXJsKQpGUk9NIGNoZWNrcwpXSEVSRSAoKG5vdygpIC0gdG9JbnRlcnZhbERheShpbnRlcnZhbF9kYXlzKSkgPD0gY2hlY2tfc3RhcnRfdGltZSkgQU5EIChwdWxsX3JlcXVlc3RfbnVtYmVyIE5PVCBJTiAoCiAgICBTRUxFQ1QgcHVsbF9yZXF1ZXN0X251bWJlciBBUyBwcm4KICAgIEZST00gY2hlY2tzCiAgICBXSEVSRSAocHJuICE9IDApIEFORCAoKG5vdygpIC0gdG9JbnRlcnZhbERheShpbnRlcnZhbF9kYXlzKSkgPD0gY2hlY2tfc3RhcnRfdGltZSkgQU5EIChjaGVja19uYW1lIElOIChiYWNrcG9ydF9hbmRfcmVsZWFzZV9zcGVjaWZpY19jaGVja3MpKQopKSBBTkQgKHBvc2l0aW9uKHRlc3RfbmFtZSwgbmFtZV9zdWJzdHIpID4gMCkgQU5EICh0ZXN0X3N0YXR1cyBJTiAoJ0ZBSUwnLCAnRVJST1InLCAnRkxBS1knKSkKR1JPVVAgQlkgZApPUkRFUiBCWSBkIERFU0MK)

https://s3.amazonaws.com/clickhouse-test-reports/65689/cfc004ee6f478716811227eaaf2caba5a81f8561/integration_tests__tsan__[5_6]//home/ubuntu/actions-runner/_work/_temp/test/output_dir/integration_run_parallel5_0.log

```
>               assert check_used_disks_with_retry(node1, name, set(["external"]), 100)
E               AssertionError: assert False
E                +  where False = check_used_disks_with_retry(<helpers.cluster.ClickHouseInstance object at 0x7fd6a92b7e20>, 'mt_test_alter_with_merge_work_1719374512', {'external'}, 100)
E                +    where {'external'} = set(['external'])
```

--- a/tests/integration/test_ttl_move/test.py
+++ b/tests/integration/test_ttl_move/test.py
@@ -76,7 +76,7 @@ def get_used_disks_for_table(node, table_name, partition=None):
     )
 
 
-def check_used_disks_with_retry(node, table_name, expected_disks, retries):
+def check_used_disks_with_retry(node, table_name, expected_disks, retries=1):
     for _ in range(retries):
         used_disks = get_used_disks_for_table(node, table_name)
         if set(used_disks).issubset(expected_disks):
@@ -1635,9 +1635,9 @@ def test_alter_with_merge_work(started_cluster, name, engine, positive):
         optimize_table(20)
 
         if positive:
-            assert check_used_disks_with_retry(node1, name, set(["external"]), 100)
+            assert check_used_disks_with_retry(node1, name, set(["external"]))
         else:
-            assert check_used_disks_with_retry(node1, name, set(["jbod1", "jbod2"]), 50)
+            assert check_used_disks_with_retry(node1, name, set(["jbod1", "jbod2"]))
 
         time.sleep(5)
 

--- a/tests/integration/test_ttl_move/test.py
+++ b/tests/integration/test_ttl_move/test.py
@@ -1613,7 +1613,7 @@ def test_alter_with_merge_work(started_cluster, name, engine, positive):
             ALTER TABLE {name} MODIFY
             TTL d1 + INTERVAL 0 SECOND TO DISK 'jbod2',
                 d1 + INTERVAL 5 SECOND TO VOLUME 'external',
-                d1 + INTERVAL 10 SECOND DELETE
+                d1 + INTERVAL 30 SECOND DELETE
         """.format(
                 name=name
             )
@@ -1647,7 +1647,7 @@ def test_alter_with_merge_work(started_cluster, name, engine, positive):
                 f"SELECT disk_name, name FROM system.parts WHERE table = '{name}' AND active = 1"
             )
 
-        time.sleep(5)
+        time.sleep(25)
 
         optimize_table(20)
 

--- a/tests/integration/test_ttl_move/test.py
+++ b/tests/integration/test_ttl_move/test.py
@@ -1635,9 +1635,17 @@ def test_alter_with_merge_work(started_cluster, name, engine, positive):
         optimize_table(20)
 
         if positive:
-            assert check_used_disks_with_retry(node1, name, set(["external"]))
+            assert check_used_disks_with_retry(
+                node1, name, set(["external"])
+            ), "Parts: " + node1.query(
+                f"SELECT disk_name, name FROM system.parts WHERE table = '{name}' AND active = 1"
+            )
         else:
-            assert check_used_disks_with_retry(node1, name, set(["jbod1", "jbod2"]))
+            assert check_used_disks_with_retry(
+                node1, name, set(["jbod1", "jbod2"])
+            ), "Parts: " + node1.query(
+                f"SELECT disk_name, name FROM system.parts WHERE table = '{name}' AND active = 1"
+            )
 
         time.sleep(5)
 

